pkg/volume/vsphere_volume/vsphere_volume_util.go

```diff
@@ -73,22 +82,63 @@ func (util *VsphereDiskUtil) CreateVolume(v *vsphereVolumeProvisioner) (vmDiskPa
-		case "diskformat":
+		case diskformat:
 			volumeOptions.DiskFormat = value
 		case "datastore":
 			volumeOptions.Datastore = value
+		case Policy_HostFailuresToTolerate:
+			if !validateVSANCapability(Policy_HostFailuresToTolerate, value) {
+				return "", 0, fmt.Errorf(`Invalid value for hostFailuresToTolerate in volume plugin %s. 
+				The default value is 1, minimum value is 0 and maximum value is 3.`, v.plugin.GetPluginName())
+			}
+			volumeOptions.StorageProfileData += " (\"hostFailuresToTolerate\" i" + value + ")"
+		case Policy_ForceProvisioning:
+			if !validateVSANCapability(Policy_ForceProvisioning, value) {
+				return "", 0, fmt.Errorf(`Invalid value for forceProvisioning in volume plugin %s. 
+				The value can be either 0 or 1.`, v.plugin.GetPluginName())
+			}
+			volumeOptions.StorageProfileData += " (\"forceProvisioning\" i" + value + ")"
+		case Policy_CacheReservation:
+			if !validateVSANCapability(Policy_CacheReservation, value) {
+				return "", 0, fmt.Errorf(`Invalid value for cacheReservation in volume plugin %s.
+				The minimum percentage is 0 and maximum percentage is 100.`, v.plugin.GetPluginName())
+			}
+			intVal, _ := strconv.Atoi(value)
+			volumeOptions.StorageProfileData += " (\"cacheReservation\" i" + strconv.Itoa(intVal*10000) + ")"
+		case Policy_DiskStripes:
+			if !validateVSANCapability(Policy_DiskStripes, value) {
+				return "", 0, fmt.Errorf(`Invalid value for diskStripes in volume plugin %s. 
+				The minimum value is 1 and maximum value is 12.`, v.plugin.GetPluginName())
+			}
+			volumeOptions.StorageProfileData += " (\"stripeWidth\" i" + value + ")"
+		case Policy_ObjectSpaceReservation:
+			if !validateVSANCapability(Policy_ObjectSpaceReservation, value) {
+				return "", 0, fmt.Errorf(`Invalid value for ObjectSpaceReservation in volume plugin %s. 
+				The minimum percentage is 0 and maximum percentage is 100.`, v.plugin.GetPluginName())
+			}
+			volumeOptions.StorageProfileData += " (\"proportionalCapacity\" i" + value + ")"
+		case Policy_IopsLimit:
+			if !validateVSANCapability(Policy_IopsLimit, value) {
+				return "", 0, fmt.Errorf(`Invalid value for iopsLimit in volume plugin %s. 
+				The value should be greater than 0.`, v.plugin.GetPluginName())
+			}
+			volumeOptions.StorageProfileData += " (\"iopsLimit\" i" + value + ")"
 		default:
 			return "", 0, fmt.Errorf("invalid option %q for volume plugin %s", parameter, v.plugin.GetPluginName())
 		}
 	}
 
+	if volumeOptions.StorageProfileData != "" {
+		volumeOptions.StorageProfileData = "(" + volumeOptions.StorageProfileData + ")"
+	}
+	glog.V(1).Infof("StorageProfileData in vsphere volume %q", volumeOptions.StorageProfileData)
 	// TODO: implement PVC.Selector parsing
 	if v.options.PVC.Spec.Selector != nil {
 		return "", 0, fmt.Errorf("claim.Spec.Selector is not supported for dynamic provisioning on vSphere")
 	}
 
 	vmDiskPath, err = cloud.CreateVolume(volumeOptions)
-	if err != nil {
+	if err != nil || vmDiskPath == "" {
```

Comment:
It would never happen. Anyways, will remove it as it would never get encountered.